File: backend/app/models/admin_navigation.py
# Copyright (c) 2026 Laurent Barbe
# Licensed under the Apache License, Version 2.0
from __future__ import annotations

from pydantic import Field

from app.models.base import ApiModel


class AdminPendingRequestCounts(ApiModel):
    identity_link_requests: int = Field(ge=0)
    portal_requests: int = Field(ge=0)

File: frontend/src/api/adminNavigation.ts
/*
 * Copyright (c) 2026 Laurent Barbe
 * Licensed under the Apache License, Version 2.0
 */
import client from "./client";

export type AdminPendingRequestCounts = {
  identity_link_requests: number;
  portal_requests: number;
};

export async function fetchAdminPendingRequestCounts(
  signal?: AbortSignal,
): Promise<AdminPendingRequestCounts> {
  const { data } = await client.get<AdminPendingRequestCounts>(
    "/admin/navigation/pending-requests",
    { signal },
  );
  return data;
}
